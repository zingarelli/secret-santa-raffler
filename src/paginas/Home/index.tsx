import Formulario from "../../componentes/Formulario/Formulario";
import ListaParticipantes from "../../componentes/ListaParticipantes";
import Rodape from "../../componentes/Rodape";

export default function Home() {
    return (
        <>
            <h2>Vamos come√ßar!</h2>
            <Formulario />
            <ListaParticipantes />
            <Rodape />
        </>
    )
}