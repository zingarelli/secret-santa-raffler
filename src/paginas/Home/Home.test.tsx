import { render } from "@testing-library/react"
import { RecoilRoot } from "recoil"
import Home from "."

// mockando o useNavigate para não precisar envolver o React Router no teste

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => jest.fn()
    }
})

describe('a página inicial', () => {
    test('deve ser renderizada corretamente', () => {
        // a propriedade container possui o DOM renderizado
        const { container } = render(
            <RecoilRoot>
                <Home />
            </RecoilRoot>
        )
        expect(container).toMatchSnapshot();
    })
});