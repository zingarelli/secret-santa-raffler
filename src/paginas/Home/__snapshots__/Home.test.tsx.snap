// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a página inicial deve ser renderizada corretamente 1`] = `
<div>
  <h2>
    Vamos começar!
  </h2>
  <form
    class="form"
  >
    <input
      class="input"
      placeholder="Insira os nomes dos participantes"
      required=""
      type="text"
      value=""
    />
    <button
      class="button"
      disabled=""
    >
      Adicionar
    </button>
  </form>
  
  <ul />
  <footer
    class="footer"
  >
    <button
      class="button--secondary"
      disabled=""
    >
      Iniciar brincadeira!
    </button>
    <img
      alt="arte mostrando sacolas de presente"
      class="image"
      src="/images/sacolas.png"
    />
  </footer>
</div>
`;
